examples/tests/nonvolatile_storage/main.c

```diff
@@ -50,6 +50,12 @@ static int test_all(void) {
   if ((r = test(readbuf, writebuf, 256, 20, 14)) != 0) return r;
   if ((r = test(readbuf, writebuf, 512, 0, 512)) != 0) return r;
 
+  printf("Write to end of region (offset %d)\n", num_bytes - 512);
+  if ((r = test(readbuf, writebuf, 512, num_bytes - 512, 500)) != 0) return r;
+  
+  printf("Write beyond end region, should fail (offset %d)\n", num_bytes);
+  if ((r = test(readbuf, writebuf, 512, num_bytes, 501)) == 0) return r;
```

Comment:
This should probably return -1 to indicate a failure rather than returning the 0 this call would have returned in the case this succeeds when it shouldn't

Reply:
Good point, that would be an error case that isn't reported.